libgadget/domain.c

```diff
@@ -48,6 +65,7 @@ typedef struct {
     int UseGlobalSort; /** Apply a global sorting on the subsamples before building the top level tree. */
     int SubSampleDistance; /** Frequency of subsampling */
     int PreSort; /** PreSort the local particles before subsampling, creating a fair subsample */
+    int NSubDomains; /** Number of sub-domain segments to create before balancing. Should be DomainOverDecompositionFactor * NTask*/
```

Comment:
I avoided using SubDomain because I thought it was confusing -- a more suitable candidate for the name is TopLeave, which is a octtree nodes augumented with workload accounting, and the spatial region element iterated by ForceTree / PM.

Segments are transient objects created only during the load balancing stage.

Reply:
`Number of Hilbert-Peano segments` rather than sub-domain segments.

Reply:
I will change the comment - I like SubDomain better than TopLeave. I always found TopLeave confusing because it sounds like it is part of a Tree, but it is not.

Reply:
Ah but it is NTopLeaves in ddecomp. I will make it NTopLeaves in policy also.

Reply:
Sorry but I think you got this wrong. In general NSegment <= NTopLeaves. Several neighbouring Leaves are combined into a segment during load balancing.

Reply:
No, I think this variable is supposed to be NTopLeaves. NSegment is still there in domain_assign_balanced, and is == NTask for now,